tardis/widgets/line_info.py

```diff
@@ -0,0 +1,749 @@
+"""Class to create and display Line Info Widget."""
+import re
+
+from astropy import units as u
+import numpy as np
+import pandas as pd
+import qgrid
+from plotly import graph_objects as go
+from plotly.callbacks import BoxSelector
+import ipywidgets as ipw
+
+from tardis.analysis import LastLineInteraction
+from tardis.util.base import species_tuple_to_string, species_string_to_tuple
+from tardis.widgets.util import create_table_widget, TableSummaryLabel
+
+
+class LineInfoWidget:
+    """
+    Widget to explore atomic lines that produced features in the simulated spectrum.
+
+    It allows selection of a wavelength range in the spectrum to display a 
+    table giving the fraction of packets that experienced their last
+    interaction with each species. Using toggle buttons, user can specify
+    whether to filter the selected range by emitted or absorbed wavelengths
+    of packets. Clicking on a row in the fractional species interactions table
+    shows packet counts for each last line interaction of the selected species,
+    which can be grouped in several ways using dropdown menu.
+    """
+
+    FILTER_MODES = ("packet_out_nu", "packet_in_nu")
+    FILTER_MODES_DESC = ("Emitted Wavelength", "Absorbed Wavelength")
+    GROUP_MODES = ("both", "exc", "de-exc")
+    GROUP_MODES_DESC = (
+        "Both excitation line (absorption) and de-excitation line (emission)",
+        "Only excitation line (absorption)",
+        "Only de-excitation line (emission)",
+    )
+    COLORS = {"selection_area": "lightpink", "selection_border": "salmon"}
+
+    def __init__(
+        self,
+        lines_data,
+        line_interaction_analysis,
+        spectrum_wavelength,
+        spectrum_luminosity_density_lambda,
+        virt_spectrum_wavelength,
+        virt_spectrum_luminosity_density_lambda,
+    ):
+        """
+        Initialize the LineInfoWidget with line interaction and spectrum data.
+
+        Parameters
+        ----------
+        lines_data : pd.DataFrame
+            Data about the atomic lines present in simulation model's plasma
+        line_interaction_analysis : dict of tardis.analysis.LastLineInteraction
+            Dictionary in which keys are the FILTER_MODES and values are the
+            LastLineInteraction objects initialized with corresponding modes
+        spectrum_wavelength : astropy.Quantity
+            Wavelength values of a real spectrum, having unit of Angstrom
+        spectrum_luminosity_density_lambda : astropy.Quantity
+            Luminosity density lambda values of a real spectrum, having unit
```

Comment:
I think we discussed before, but the units should probably be written as 'erg / s' rather than 'Hz erg'

Reply:
Yes this was written way earlier before I changed the units, good catch - fixed it now!